utils/keytar.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as keytarType from 'keytar';
+import { getCoreNodeModule } from './getCoreNodeModule';
+
+export interface IKeytar {
+    /**
+     * Get the stored password for the service and account.
+     *
+     * @param service The string service name.
+     * @param account The string account name.
+     *
+     * @returns A promise for the password string.
+     */
+    getPassword(service: string, account: string): Promise<string | null>;
+
+    /**
+     * Add the password for the service and account to the keychain.
+     *
+     * @param service The string service name.
+     * @param account The string account name.
+     * @param password The string password.
+     *
+     * @returns A promise for the set password completion.
+     */
+    setPassword(service: string, account: string, password: string): Promise<void>;
+
+    /**
+     * Delete the stored password for the service and account.
+     *
+     * @param service The string service name.
+     * @param account The string account name.
+     *
+     * @returns A promise for the deletion status. True on success.
+     */
+    deletePassword(service: string, account: string): Promise<boolean>;
+}
+
+/**
+ * Returns the keytar module installed with vscode
+ */
+function getKeytarModule(): typeof keytarType {
+    const keytar: typeof keytarType | undefined = getCoreNodeModule('keytar');
+    if (!keytar) {
+        throw new Error("Internal error: Could not find keytar module for reading and writing passwords");
```

Comment:
Unless you can prove this extension is useless without keytar, I think we should keep it optional (aka don't throw an error here and change the type to `| undefined` in `ext` like it was before your changes). We're relying on something shipped with VS Code and I just don't think we can assume it will always be there (I mean they've already changed it once).